[CI] Clang-format and checkstyle failures lack actionable detail in job logs

## Problem

Two of the FE/BE lint CI jobs hide actionable failure detail in the GitHub PR UI and emit nothing useful in the workflow job log itself. This forces contributors to switch tabs and click through the PR Files-changed view to figure out what to fix — and on fork PRs the information is sometimes lost entirely.

### 1. `ci-clang-format.yml` — silent on fork-PR auto-push failures

The workflow runs `clang-format` over the diff and tries to push a fixup commit back into the PR head branch. On fork PRs the push usually fails because the workflow bot has no write access to the fork (this is the common case for any external contributor, including all of ours).

The fail-path emits only:
```
::error::Auto-push failed. Files that need formatting:
<git diff --name-only>
::error::To fix locally: CLANG_FORMAT_BINARY=clang-format-10 bash build-support/clang-format.sh
```

So the log tells you *which* files have formatting issues but never *what* the diff is. To reproduce locally you need `clang-format-10` specifically — which is hard to come by on modern dev environments (Homebrew on macOS only ships clang-format from v11 upwards, and the major distros have moved past 10 as well). That makes "just run it locally" a non-trivial setup ask just to see what one space the bot wants to change.

### 2. `ci-pipeline*.yml` checkstyle step — no errors in job log

Both `ci-pipeline.yml` and `ci-pipeline-branch.yml` invoke `dbelyaev/action-checkstyle@v1.15.0` with `reporter: "github-pr-check"`. With that reporter, reviewdog publishes findings only to the GitHub Check Runs API — they appear as annotations on the PR Files-changed tab. The job log itself contains no `File.java:42:5: error: ...` lines, just a pass/fail summary.

This used to be much more informative — at some point in the past, checkstyle errors were printed directly to the job log, and you could triage from the Actions tab alone. Today, looking at a failed checkstyle job log gives you essentially zero information about what failed.

## Expected

- clang-format fail-path on fork PR should print the unified diff in the job log (collapsible group preferable, copy-paste applicable via `git apply`).
- Checkstyle step should keep its PR annotations but also tee the findings to stdout so they're visible in the job log.

## Reproduce

Open any PR from a fork (no "Allow edits from maintainers") that introduces a formatting issue → Clang-Format job fails with file names but no diff.

For checkstyle, intentionally break a checkstyle rule in any `.java` file under `fe/` → Run Java checkstyle step fails with no log detail; only the Files-changed annotation shows the issue.

--- a/.github/workflows/ci-clang-format.yml
+++ b/.github/workflows/ci-clang-format.yml
@@ -174,6 +174,9 @@ jobs:
             git reset HEAD~1
             echo "::error::Auto-push failed. Files that need formatting:"
             git diff --name-only
+            echo "::group::Full clang-format diff (apply locally with 'git apply' to fix)"
+            git --no-pager diff
+            echo "::endgroup::"
             if [ "${HEAD_REPO}" != "${{ github.repository }}" ]; then
               echo "::error::This is a fork PR. Please enable 'Allow edits from maintainers' on your PR."
             else

--- a/.github/workflows/ci-pipeline-branch.yml
+++ b/.github/workflows/ci-pipeline-branch.yml
@@ -464,6 +464,7 @@ jobs:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           fail_on_error: true
           level: error
+          reviewdog_flags: "-tee"
 
   fe-ut:
     runs-on: [self-hosted, normal]

--- a/.github/workflows/ci-pipeline.yml
+++ b/.github/workflows/ci-pipeline.yml
@@ -765,6 +765,7 @@ jobs:
           github_token: ${{ secrets.GITHUB_TOKEN }}
           fail_on_error: true
           level: error
+          reviewdog_flags: "-tee"
 
   fe-ut:
     runs-on: [self-hosted, normal]

